lib/spack/spack/package.py

```diff
@@ -1406,22 +1397,6 @@ def virtuals_provided(self):
         return [vspec for vspec, constraints in self.provided.items()
                 if any(self.spec.satisfies(c) for c in constraints)]
 
-    @property
-    def installed(self):
-        """Installation status of a package.
-
-        Returns:
-            True if the package has been installed, False otherwise.
-        """
-        try:
-            # If the spec is in the DB, check the installed
-            # attribute of the record
-            return spack.store.db.get_record(self.spec).installed
-        except KeyError:
-            # If the spec is not in the DB, the method
-            #  above raises a Key error
-            return False
```

Comment:
I suspect a lot of downstream code (spackdev, spack-manager, scripts, etc.) uses these methods, or at least `installed()`. We should add stubs for both of these that work by delegating to the spec, but print deprecation warnings, and remove them in 0.19.

I wonder if it’s worth having some kind of test decorator so we could mark the tests for these to fail after a certain version to remind us to remove them.

Reply:
Fixed in f862dce